src/libs/ajax/data-table-providers/WdsDataTableProvider.test.ts

```diff
@@ -0,0 +1,302 @@
+    return this.transformPage(wdsPage, recordType, queryOptions)
+  }
+}
+
+const recordType: string = 'mytype'
+
+const queryOptions: EntityQueryOptions = {
+  pageNumber: 2,
+  itemsPerPage: 50,
+  sortField: 'stringAttr',
+  sortDirection: 'desc',
+  snapshotName: '',
+  googleProject: '',
+  activeTextFilter: '',
+  filterOperator: ''
+}
+
+
+describe('WdsDataTableProvider', () => {
+  let getRecords
+  let deleteTable
+  let downloadTsv
+
+  beforeEach(() => {
+    getRecords = jest.fn().mockImplementation((_instanceId: string, _recordType: string, _parameters: SearchRequest) => {
+      const recordQueryResponse: RecordQueryResponse = {
+        searchRequest: {
+          limit: 10,
+          offset: 0,
+          sort: 'desc',
+          sortAttribute: 'numericAttr'
+        },
+        records: [
+          {
+            id: '2',
+            type: 'item',
+            attributes: {
+              arrayBoolean: [
+                true,
+                false
+              ],
+              arrayDate: [
+                '2022-11-03'
+              ],
+              arrayDateTime: [
+                '2022-11-03T04:36:20'
+              ],
+              arrayNumber: [
+                12821.112,
+                0.12121211,
+                11
+              ],
+              arrayString: [
+                'green',
+                'red'
+              ],
+              booleanAttr: true,
+              numericAttr: 2,
+              stringAttr: 'string'
+            }
+          },
+          {
+            id: '1',
+            type: 'item',
+            attributes: {
+              arrayBoolean: [
+                true,
+                false
+              ],
+              arrayDate: [
+                '2022-11-03'
+              ],
+              arrayDateTime: [
+                '2022-11-03T04:36:20'
+              ],
+              arrayNumber: [
+                12821.112,
+                0.12121211,
+                11
+              ],
+              arrayString: [
+                'green',
+                'red'
+              ],
+              booleanAttr: true,
+              numericAttr: 1,
+              stringAttr: 'string'
+            }
+          }
+        ],
+        totalRecords: 2
+      }
+      return Promise.resolve(recordQueryResponse)
+    })
+    deleteTable = jest.fn().mockImplementation((_instanceId: string, _recordType: string) => {
+      return Promise.resolve(new Response('', { status: 204 }))
+    })
+    downloadTsv = jest.fn().mockImplementation((_instanceId: string, _recordType: string) => {
+      return Promise.resolve(new Blob(['hello']))
+    })
+
+    asMockedFn(Ajax).mockImplementation(() => ({ WorkspaceDataService: { getRecords, deleteTable, downloadTsv } } as ReturnType<typeof Ajax>))
+  })
+
+  describe('transformPage', () => {
+    it('restructures a WDS response', () => {
+      // ====== Arrange
```

Comment:
The agreed standard on these headers is 
// Arrange
// Act
// Assert

At other companies, I've gone with ALL CAPS version of that, but above is where we landed here.  A blank line between the sections helps break it up too.